app/app.go

```diff
@@ -279,8 +278,7 @@ func New(
 	for _, h := range cast.ToIntSlice(appOpts.Get(server.FlagUnsafeSkipUpgrades)) {
 		skipUpgradeHeights[int64(h)] = true
 	}
-	homePath := cast.ToString(appOpts.Get(flags.FlagHome))
-	app.UpgradeKeeper = upgradekeeper.NewKeeper(skipUpgradeHeights, runtime.NewKVStoreService(keys[upgradetypes.StoreKey]), encodingConfig.Codec, homePath, app.BaseApp, govModuleAddr)
+	app.UpgradeKeeper = upgradekeeper.NewKeeper(skipUpgradeHeights, runtime.NewKVStoreService(keys[upgradetypes.StoreKey]), encodingConfig.Codec, DefaultNodeHome, app.BaseApp, govModuleAddr)
```

Comment:
Can `DefaultNodeHome` still be set by the custom home flag?

Reply:
Yes client/v2 has some helpers that does some [early flag parsing](https://github.com/cosmos/cosmos-sdk/blob/2c43a575be1b8c0ed41cf6ae003f8236836b3e68/client/v2/helpers/home.go#L13-L48): https://github.com/celestiaorg/celestia-app/blob/13b8ff2cad4a094ea7464471d9b3e45c24dc0263/app/init.go#L26-L35